app/controllers/spree/orders_controller.rb

```diff
@@ -26,8 +26,14 @@ class OrdersController < ::BaseController
 
     def show
       @order = Spree::Order.find_by!(number: params[:id])
-      ProcessPaymentIntent.new(params["payment_intent"], @order).call!
-      @order.reload
+
+      if params.key?("payment_intent")
+        result = ProcessPaymentIntent.new(params["payment_intent"], @order).call!
+        unless result.ok?
+          flash[:error] = "#{I18n.t("payment_could_not_process")}. #{result.error}"
+        end
+        @order.reload
+      end
```

Comment:
I plan to move this to its own purpose-specific controller action in an upcoming PR so we don't mix the two concerns and potentially introduce bugs.